addons/mrp/models/stock_rule.py

```diff
@@ -143,7 +143,10 @@ def _get_matching_bom(self, product_id, company_id, values):
             return values['bom_id']
         if values.get('orderpoint_id', False) and values['orderpoint_id'].bom_id:
             return values['orderpoint_id'].bom_id
-        return self.env['mrp.bom']._bom_find(product_id, picking_type=self.picking_type_id, bom_type='normal', company_id=company_id.id)[product_id]
+        bom = self.env['mrp.bom']._bom_find(product_id, picking_type=self.picking_type_id, bom_type='normal', company_id=company_id.id)[product_id]
+        if bom:
+            return bom
+        return self.env['mrp.bom']._bom_find(product_id, picking_type=False, bom_type='normal', company_id=company_id.id)[product_id]
```

Comment:
You will also need to update the `mrp.bom`'s `picking_type_id` help string, as it's not really behaving as it's told there anymore. :slightly_smiling_face: